SoObjects/SOGo/SOGoGroup.m

```diff
@@ -108,13 +109,29 @@ - (void) dealloc
 + (id) groupWithIdentifier: (NSString *) theID
                   inDomain: (NSString *) domain
 {
-  NSString *uid;
+  NSRange r;
+  NSString *uid, *inDomain;
+  SOGoSystemDefaults *sd;
 
   uid = [theID hasPrefix: @"@"] ? [theID substringFromIndex: 1] : theID;
+  inDomain = domain;
+
+  sd = [SOGoSystemDefaults sharedSystemDefaults];
+  if ([sd enableDomainBasedUID])
+    {
+      /* Split domain from uid */
+      r = [uid rangeOfString: @"@" options: NSBackwardsSearch];
+      if (r.location != NSNotFound)
+        {
+          uid = [uid substringToIndex: r.location];
+          if (!domain)
+            inDomain = [uid substringFromIndex: r.location + 1];
```

Comment:
You already trimmed the `uid` variable, so the characters from position `r + 1`are gone.